### PR TITLE
Fix nesting operations issue

### PR DESCRIPTION
**Pull request by Acentelles**
Simple operations like
```
sig main : field -> field -> field -> field
let main x y z = x + y
```
compiled, whereas more complex (nested) ones didn't
```
sig main : field -> field -> field -> field
let main x y z = x^2 + y^2 + z^2
```

This fix allows to compile an arbitrary number or operations.
See https://github.com/anoma/juvix/pull/900/files#diff-89b66698a8860c8b48518210cac3b4a57140cb82355d5240a8235fb189958ae6 as an example file that compiles.

See https://github.com/anoma/juvix/pull/750 for reference.

### CODE REVIEW

File: library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs
Comment (mariari) [inline library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs:160]: I'm guessing we can do this more minimal version as we keeping the same extension rather than changing it?

File: library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs
Comment (mariari) [inline library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs:178]: Seems we have this function here as well!

File: library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs
Comment (mariari) [inline library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs:283]: Are these related to the fix at all?

File: library/Core/src/Juvix/Core/Erased/Ann/Conversion.hs
Comment (mariari) [inline library/Core/src/Juvix/Core/Erased/Ann/Conversion.hs:209]: This function shows up again. 

We should unify this with the other one, as it's hard to maintain copies.

Also in this case, could we pass in an extra usage like we did with the other one, or do we not have that in scope anywhere?

File: library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs
Comment (mariari) [inline library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs:241]: Seems this function also shows up here as well

File: library/Core/src/Juvix/Core/IR/Evaluator/SubstV.hs
Comment (mariari) [inline library/Core/src/Juvix/Core/IR/Evaluator/SubstV.hs:422]: We should probably note where all these unimplemented parts are and try to see if we can centralize this notion, as it seems to spread out all over the place, which makes maintaining harder.

**Review by mariari — CHANGES_REQUESTED** [review]
It seems overall this PR does fix the nesting issues, however it's hard to see exactly how with all the other unreleated changes along with them.

Also it seems the conversion code is spread out between many different modules with varying levels of comments about baked in behavior.

Would be cool if we had a single module that had this change along with a variant that took a usage or non at all

File: library/Core/src/Juvix/Core/IR/Evaluator/SubstV.hs
Comment (Acentelles) [reply library/Core/src/Juvix/Core/IR/Evaluator/SubstV.hs:423]: The `notImplemented` bits have to do with handling `Cont`. Could we leave the implementation of those to a separate PR?

Comment (mariari) [reply library/Core/src/Juvix/Core/IR/Evaluator/SubstV.hs:423]: I'll make a PR myself after merging this, so I don't lose where this particular issue scatters to

**Review by mariari — APPROVED** [review]
I will merge this come monday after merging develop

**Pull request merged**
